x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/FrozenCacheService.java

```diff
@@ -271,24 +255,42 @@ private long getRegionSize(long fileLength, int region) {
         return effectiveRegionSize;
     }
 
+    /**
+     * Returns a {@link CacheFileRegion} instance for the given file parameters. The returned instance is reference must have its
+     * reference count decremented by one once no longer used.
```

Comment:
Some of the Javadocs read a bit funny (missing / extra words). Can you fix them?